### ISSUE TITLE
Document class swCarbon

### ISSUE THREAD

**Issue by Zachary-Kramer** [Burke-Lauenroth-Lab/rSOILWAT2#79]
The class and its methods need formal roxygen2 documentation. Note that other classes are not documented, so I'm not sure what our 'best practices' are for the verbosity of our documentation, so I will do my best and follow roxygen2 guidelines.

**Comment by Zachary-Kramer** [comment]
I originally opted for a very verbose documentation of all of the generic methods, but the documentation was unnecessarily complex for the user. So, I reiterated through the documentation and squashed everything into the swCarbon class. I have some cleaning up to do (such as the title of the documentation being incorrect), but for the most part it is complete. @dschlaep is the **approach** I took below acceptable?

![image](https://user-images.githubusercontent.com/16931318/32996697-4cd820c8-cd86-11e7-905e-3f16bd10b118.png)
![image](https://user-images.githubusercontent.com/16931318/32996713-7ab9c730-cd86-11e7-8983-9ee16ad305f3.png)

**Comment by dschlaep** [comment]
Yes, most/all classes/methods are not documented, but should eventually be documented. I think your suggestion above looks sufficient for `swCarbon`.

For other more complicated cases, we should add 'note' sections that explain in more detail what individual methods are doing.

I have a few questions/remarks based on commit eea15d440d323f3790fb492a19ca2f8d625b002e:
- It appears that you didn't use the tag `@slot` to describe slots as described [in the section 'Documenting classes, generics and methods' by the roxygen2 vignette](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html).

- I don't understand why you use both the `@name` and `@rdname` tags for each documentation object [see 'Documenting multiple functions in the same file'](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html).

- Why did you switch from `@export` to `@exportMethod` and `@exportClass`? The [vignette](https://cran.r-project.org/web/packages/roxygen2/vignettes/namespace.html) seems to suggest that `@export` is sufficient.

- Why are four documentation files in the main folder and not in the subfolder `man/` where all other `.Rd` files are located: swCarbon-class.Rd, swCarbon-methods.Rd, swInputData-class.Rd, swInputData-methods.Rd?

- The documentation of class `swCarbon` states "The generic functions listed work on this and the proper sw Class in the container's slots." This is not true (unlike for most/all other methods/classes, e.g., `swProd_Albedo`). The carbon-related methods have been implemented only for signature `swInputData` but not for their own class `swCarbon`.
```
path_demo3 <- system.file("extdata", "example3", package = "rSOILWAT2")
sw_in6 <- sw_inputDataFromFiles(dir = path_demo3, files.in = "files_v31.in")
class(sw_in6)
```
> [1] "swInputData"
```
co2_in6 <- get_swCarbon(sw_in6)
class(co2_in6)
```
> [1] "swCarbon"
```
swCarbon_Use_Bio(sw_in6)
```
> [1] 1
```
swCarbon_Use_Bio(co2_in6)
```
> Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘swCarbon_Use_Bio’ for signature ‘"swCarbon"’

For instance, `swProd_Albedo` has been implemented for signature `swInputData` and `swProd`:
```
methods("swProd_Albedo")
```
> [1] swProd_Albedo,swInputData-method swProd_Albedo,swProd-method
```
swProd_Albedo(sw_in6)
swProd_Albedo(slot(sw_in6, "prod"))
```
> Grasses      Shrubs       Trees       Forbs Bare Ground 
> 0.167       0.143       0.106       0.167       0.150 
> Grasses      Shrubs       Trees       Forbs Bare Ground 
> 0.167       0.143       0.106       0.167       0.150

**Comment by Zachary-Kramer** [comment]
> It appears that you didn't use the tag @slot to describe slots as described

It looks like I was using guidelines for Roxygen2 3.0.0, not the newest version. It seems they updated the way they want you to document slots.

> I don't understand why you use both the @name and @rdname tags for each documentation object

The references I used do @name and @rdname for classes, and I seem to have missed that the they only used @rdname for methods

> Why did you switch from @export to @exportMethod and @exportClass? The vignette seems to suggest that @export is sufficient.

Again, this is apparently because I was using older documentation.

> Why are four documentation files in the main folder and not in the subfolder man/ where all other .Rd files are located: swCarbon-class.Rd, swCarbon-methods.Rd, swInputData-class.Rd, swInputData-methods.Rd?

I accidentally committed them. I made copies of those files to diagnose an issue but apparently forgot to delete them.

> The documentation of class swCarbon states "The generic functions listed work on this and the proper sw Class in the container's slots." This is not true (unlike for most/all other methods/classes, e.g., swProd_Albedo). The carbon-related methods have been implemented only for signature swInputData but not for their own class swCarbon.

Ok, good to know. I will write the same functions for swCarbon.

I will clean all of this up after solving the higher priority issue #78. Thanks for taking a deep look at this.

**Comment by dschlaep** [comment]
The tag `@slot` has been around since roxygen2 v3.0.0 which was published in 2013 (https://github.com/klutometis/roxygen/releases/tag/v3.0.0)

Thanks for cleaning up. 

For information: the book [R packages](http://r-pkgs.had.co.nz/) by Hadley Wickham is always a good source of information in general (which our [updated README](https://github.com/Burke-Lauenroth-Lab/rSOILWAT2/tree/Rsoilwat_co2_effects#for-code-contributors-only) cites frequently). The relevant chapter for [documentation is here](http://r-pkgs.had.co.nz/man.html) which in this case is about the same as the vignettes for the `roxygen2` package that I referenced earlier.

**Comment by Zachary-Kramer** [comment]
I've finished implementing your feedback. One thing I noticed is that the methods must use @name (not @rdname), otherwise the search results are confusing. For instance, when I search only with @rdname for `??swCarbon`, the title of the documentation will be the name of the first method defined for that class, which is in this case swCarbon_Use_Bio:

![image](https://user-images.githubusercontent.com/16931318/33527434-48ad1d02-d851-11e7-8f04-67b3b233017c.png)

So, @name must be used to create the alias `swCarbon-class` for the first defined method of the class to achieve results like this:

![image](https://user-images.githubusercontent.com/16931318/33527505-75310d74-d852-11e7-8779-f5cf55f8d3f3.png)

For consistency and being less prone to problems when people move code around, all the methods use @name for either swCarbon or swInputData.
